drivers/serial/Kconfig.stm32

```diff
@@ -3,10 +3,15 @@
 # Copyright (c) 2016 Open-RnD Sp. z o.o.
 # SPDX-License-Identifier: Apache-2.0
 
+DT_COMPAT_STM32_DMA := st,stm32-dma
+
 config UART_STM32
 	bool "STM32 MCU serial driver"
 	select SERIAL_HAS_DRIVER
 	select SERIAL_SUPPORT_INTERRUPT
+	# the ASYNC implementation requires a DMA controller
+	select SERIAL_SUPPORT_ASYNC if $(dt_compat_enabled,$(DT_COMPAT_STM32_DMA))
```

Comment:
`st,stm32-dma` could be enabled for other purpose (eg: spi), so we can't assume SERIAL_SUPPORT_ASYNC=y only based on `st,stm32-dma` status.

Reply:
Thanks for the reply! I am not sure I understand. `SERIAL_SUPPORT_ASYNC` only states that the async api is supported and not that it is enabled. For that one needs to enable `UART_ASYNC_API`. This is how it was done in `Kconfig.sam0`

Reply:
> SERIAL_SUPPORT_ASYNC only states that the async api is supported and not that it is enabled. For that one needs to enable UART_ASYNC_API. 

Problem remains. Let's say board dts contains a dma node enabled because dma is used on a spi node (or qspi). This doesn't mean dmas prop is set for a uart node and that compilation will be fine.
So I'd prefer to have `SERIAL_SUPPORT_ASYNC` set unconditionally and instead work on a meaningful error is node is misconfigured.
```
select SERIAL_SUPPORT_ASYNC
```